### DISCUSSION TITLE
How to broadcast but exclulde specific Sids?

### DISCUSSION THREAD

**Discussion by hlf20010508** [Totodore/socketioxide#323] [Q&A] [answered]
I checked the code of `broadcast`, it seems that it will emit to the sender too.
The `except` need a `room` parameter, but my client don't use room.
Is there any simple way to exclude some Sids (or sender itself)?

**Comment by Totodore** [comment, marked as answer]
Rooms are only a server thing. You can just join your socket to its corresponding room sid at connection (`socket.join(socket.sid);`).

Later you can do:
```rust
socket.broadcast().except(sid1, sid2);
```

Also not that the broadcast [operator exclude the sender by default.](https://docs.rs/socketioxide/latest/socketioxide/operators/struct.BroadcastOperators.html#method.broadcast)
